server.go

```diff
@@ -52,48 +54,36 @@ type Server struct {
 	watchers     map[chan events.Event]struct{}
 	watchersLock sync.RWMutex
 
-	versionInfo        internal.VersionInfo
+	// storeBuilder builds message stores.
+	storeBuilder store.Builder
+
+	// cmdExecProfBuilder builds command profiling collectors.
 	cmdExecProfBuilder profiling.CmdProfilerBuilder
 
-	storeBuilder store.StoreBuilder
-	dataPath     string
+	// versionInfo holds info about the Gluon version.
+	versionInfo internal.VersionInfo
 }
 
 // New creates a new server with the given options.
-// It stores data in the given directory.
-func New(dir string, withOpt ...Option) (*Server, error) {
-	backend, err := backend.New(dir)
+func New(withOpt ...Option) (*Server, error) {
+	builder, err := newBuilder()
 	if err != nil {
 		return nil, err
 	}
 
-	server := &Server{
-		backend:            backend,
-		listeners:          make(map[net.Listener]struct{}),
-		sessions:           make(map[int]*session.Session),
-		watchers:           make(map[chan events.Event]struct{}),
-		cmdExecProfBuilder: &profiling.NullCmdExecProfilerBuilder{},
-		storeBuilder:       &store.OnDiskStoreBuilder{},
-		dataPath:           os.TempDir(),
-	}
-
 	for _, opt := range withOpt {
-		opt.config(server)
-	}
-
-	if err := os.MkdirAll(server.dataPath, 0o700); err != nil {
-		return nil, err
+		opt.config(builder)
 	}
 
-	return server, nil
+	return builder.build()
 }
 
 // AddUser creates a new user and generates new unique ID for this user. If you have an existing userID, please use
 // LoadUser instead.
 func (s *Server) AddUser(ctx context.Context, conn connector.Connector, encryptionPassphrase []byte) (string, error) {
 	userID := s.backend.NewUserID()
 
-	if err := s.LoadUser(ctx, conn, userID, encryptionPassphrase); err != nil {
+	if err := s.LoadUser(ctx, conn, s.backend.NewUserID(), encryptionPassphrase); err != nil {
```

Comment:
This change coupled with the next in `LoadUser` is not allowing the specified userID to be used.

Reply:
Good catch, must have pressed "undo" one too few times :smile: fixed